File: practicas/practica2_816624_845154/java/Main.java
/*
 Archivo: Main.java
 Autores: Alejandro López 845154, Adrián Pérez 816624
 Fecha últ modificación: 27/02/2024
 */
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Stack<Integer> pila = new Stack<>();

        Sumador sumador = new Sumador();
        sumador.listar();
        System.out.println("\n");
        sumador.ejecutar(pila);
        System.out.println("\n");

        CuentaAtras cuenta = new CuentaAtras();
        cuenta.listar();
        System.out.println("\n");
        cuenta.ejecutar(pila);
        System.out.println("\n");

        Factorial factorial = new Factorial();
        factorial.listar();
        System.out.println("\n");
        factorial.ejecutar(pila);
    }
}

File: practicas/practica2_816624_845154/c++/main.cc
/*
 * Programa principal que ejecuta las operaciones de la pila.
 * Autores: Alejandro López 845154, Adrián Pérez 816624
 * Fecha últ. modificación: 27/02/2024
 */
#include <iostream>
#include <stack>

#include "programa.h"

int main() {
    std::stack<int> pila;

    Sumador sumador;
    sumador.listar();
    std::cout << "\n";
    sumador.ejecutar(pila);
    std::cout << "\n";

    CuentaAtras cuenta;
    cuenta.listar();
    std::cout << "\n";
    cuenta.ejecutar(pila);
    std::cout << "\n";

    Factorial factorial;
    factorial.listar();
    std::cout << "\n";
    factorial.ejecutar(pila);

    return 0;
}